File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        unordered_map<char, int> stv = {
            { 'I' , 1 },
            { 'V' , 5 },
            { 'X' , 10 },
            { 'L' , 50 },
            { 'C' , 100 },
            { 'D' , 500 },
            { 'M' , 1000 }
        };
        
        int res = stv[s.back()];
        
        for(int i = s.length() - 2 ; i >= 0 ; i--){
            if( stv[s[i]] < stv[s[i + 1]]){
                res -= stv[s[i]];
            } else {
                res += stv[s[i]];
            }
        }
        
        return res;
    }
};

File: 0013-roman-to-integer/0013-roman-to-integer.go
func romanToInt(s string) int {
    m := make(map[byte]int)
    m['I'] = 1
    m['V'] = 5
    m['X'] = 10
    m['L'] = 50
    m['C'] = 100
    m['D'] = 500
    m['M'] = 1000

    ans := 0

    for i := 0 ; i < len(s) ; i++ {
        if i + 1 < len(s) && m[s[i]] < m[s[i + 1]] {
            ans -= m[s[i]]
        } else {
            ans += m[s[i]]
        }
    }
    return ans
}